.github/workflows/ci.yml

```diff
@@ -64,31 +65,42 @@ jobs:
         with:
           version: latest
           args: buildAll
+  e2e:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: npm install -g yarn && yarn
+
+      - name: Install Playwright Browsers
+        run: yarn playwright install --with-deps
+
+      - name: Start containers
+        shell: bash
+        run: |
+          docker compose pull
+          docker compose up --detach --wait --wait-timeout 60
 
-      # - name: Check for E2E
-      #   id: check-for-e2e
-      #   run: |
-      #     if [ -d "cypress" ]
-      #     then
-      #       echo "has-e2e=true" >> $GITHUB_OUTPUT
-      #     fi
-
-      # - name: Start grafana docker
-      #   if: steps.check-for-e2e.outputs.has-e2e == 'true'
-      #   run: docker-compose up -d
-
-      # - name: Run e2e tests
-      #   if: steps.check-for-e2e.outputs.has-e2e == 'true'
-      #   run: yarn e2e
-
-      # - name: Stop grafana docker
-      #   if: steps.check-for-e2e.outputs.has-e2e == 'true'
-      #   run: docker-compose down
-
-      # - name: Archive E2E output
-      #   uses: actions/upload-artifact@v4
-      #   if: steps.check-for-e2e.outputs.has-e2e == 'true' && steps.run-e2e-tests.outcome != 'success'
-      #   with:
-      #     name: cypress-videos
-      #     path: cypress/videos
-      #     retention-days: 5
+      - name: Load SurrealDB data
+        shell: bash
+        run: |
+          curl -L "https://datasets.surrealdb.com/surreal-deal-v1.surql" -o surreal-deal-v1.surql
+          curl -v -X POST -u "root:test" -H "NS: test" -H "DB: test" -H "Accept: application/json" --data-binary @surreal-deal-v1.surql http://127.0.0.1:8000/import
+
+      - name: Run Playwright tests
+        run: yarn playwright test
+
+      - uses: actions/upload-artifact@v4
+        if: always()
```

Comment:
Is this `if` needed?

Reply:
Yes, it is needed to ensure the step runs even if the previous steps have failed.